tools/ingest.js

```diff
@@ -0,0 +1,238 @@
+ *
+ * make sure that you have the correct path to your jadecli install, if you have it aliased in your bashrc it should be the same path as there.
+ * also, make sure the source and target environments are the names of the DATA projects!!
+ *
+ * you may need to give yourself permissions to upload to the jade-testdata bucket as well
+ */
+
+const https = require('https');
+const _ = require('lodash');
+const fs = require('fs');
+const axios = require('axios');
+const { exec } = require('child_process');
+const inquirer = require('inquirer');
+const { argv } = require('yargs').command(
+  'ingest',
+  'copy a subset of production data into another environment',
+  (args) =>
+    args
+      .positional('datasetName', {
+        describe: 'name of the dataset to rolldown',
+        type: 'string',
+      })
+      .option('source-url', {
+        alias: 'sourceUrl',
+        desc: 'url for source environment',
+        default: 'broad-jade-dev-data',
+        type: 'string',
+      })
+      .option('target-url', {
+        alias: 'targetUrl',
+        desc: 'url for target environment',
+        default: 'broad-jade-dev-data',
+        type: 'string',
+      })
+      .option('target-env', {
+        alias: 'targetEnv',
+        desc: 'google (data) project for target environment',
+        default: 'dev',
+        type: 'string',
+      })
+      .option('src-env', {
+        alias: 'srcEnv',
+        desc: 'google (data) project for source environment',
+        default: 'dev',
+        type: 'string',
+      })
+      .option('target-name', {
+        alias: 'targetName',
+        desc: 'target dataset name',
+        default: 'datasetName',
+        type: 'string',
+      })
+      .option('jc-path', {
+        alias: 'jc',
+        desc: 'path to run jadecli',
+        default: '',
+        type: 'string',
+      })
+      .option('bucket-name', {
+        alias: 'bucketName',
+        desc: 'name of the bucket to place in jade-testdata',
+        default: '',
+        type: 'string',
+      })
+      .option('verbose', {
+        alias: 'v',
+        desc: 'print extra information',
+        default: false,
+        type: 'boolean',
+      }),
+);
+
+const CORE_BILLING_ACCOUNT_ID = '00708C-45D19D-27AAFA';
+
+if (argv.verbose) {
+  axios.interceptors.request.use((request) => {
+    console.log('Starting Request', request);
+    return request;
+  });
+}
+
+const agent = new https.Agent({
+  // rejectUnauthorized: false,
+});
+
+function call(shell) {
+  console.info(`running ${shell}:`);
+  return new Promise((resolve) => {
+    exec(shell, (err, data, stderr) => {
+      if (err) {
+        console.error(err);
+      }
+      if (stderr) {
+        console.error(stderr);
+      }
+      resolve(data);
+    });
+  });
+}
+
+function switchAccount(email) {
+  return new Promise((resolve) => {
+    call(`gcloud config set account ${email} && gcloud auth print-access-token`).then((data) => {
```

Comment:
Hrm, do we need to do some extra validation here to make sure they can't pass in something like `person@gmail.com; cat /etc/password`?